tests/unit/lxd/test_lxd_instance.py

```diff
@@ -841,10 +864,107 @@ def test_unmount_error(mock_lxc, instance):
     assert exc_info.value == LXDError(
         brief=(
             "Failed to unmount 'not-mounted'"
-            " in instance 'test-instance' - no such disk."
+            f" in instance '{_TEST_INSTANCE['instance-name']}' - no such disk."
         ),
         details=(
             "* Disk device configuration: {'disk-/target':"
             " {'path': '/target', 'source': '/source', 'type': 'disk'}}"
         ),
     )
+
+
+@pytest.mark.parametrize(
+    "name",
+    [
+        "t",
+        "test",
+        "test1",
+        "test-1",
+        "this-is-40-characters-xxxxxxxxxxxxxxxxxX",
+    ],
+)
+def test_set_instance_name_unchanged(mock_lxc, name):
+    """Verify names that are already compliant are not changed."""
+    instance = LXDInstance(
+        name=name,
+        lxc=mock_lxc,
+    )
+
+    assert instance.instance_name == name
+
+
+@pytest.mark.parametrize(
+    "name, expected_name",
+    [
+        # trim away invalid beginning characters
+        ("1test", "test"),
+        ("123test", "test"),
+        ("-test", "test"),
+        ("1-2-3-test", "test"),
+        # trim away invalid ending characters
+        ("test-", "test"),
+        ("test--", "test"),
+        ("test1-", "test1"),
+        # trim away invalid characters
+        ("test$", "test"),
+        ("test-!@#$%^&*()test", "test-test"),
+        ("$1test", "test"),
+        ("test-$", "test"),
+        # truncate at 40 characters
+        (
+            "this-is-41-characters-xxxxxxxxxxxxxxxxxXx",
```

Comment:
this is what I meant above, this is a totally valid name!

Reply:
Whoops, thanks for catching.  This was an accidental oversight.  Fixed!